### ISSUE TITLE
Opens file in unexpected split

### ISSUE THREAD

**Issue by exchgr** [rking/ag.vim#92]
When I run `:Ag` while focused on a split, I expect a result to be opened in that same split. However, it seems to pick a split at random, which is kind of disruptive.

**Comment by losingkeys** [comment]
`:Ag` opens the [quickfix window](http://vimdoc.sourceforge.net/htmldoc/quickfix.html#quickfix-window) (`:LAg` opens the [location list](http://vimdoc.sourceforge.net/htmldoc/quickfix.html#location-list-window)). These might open in strange spots depending on what you (or other plugins) have set ['splitbelow'](http://vimdoc.sourceforge.net/htmldoc/options.html#'splitbelow') or ['splitright'](http://vimdoc.sourceforge.net/htmldoc/options.html#'splitright') to.

**Comment by exchgr** [comment]
Sorry, I don't think I explained this so well. The quick fix window opens in the correct position (the bottom of the current window). The issue I'm having is when I already have several splits open. The cursor will be focused on one of these splits. 

When I run `:Ag <search term>`, here's what I expect to happen:

1. Ag.vim opens a quickfix window on the bottom of the current window.
1. Nothing else happens until I pick a file.
1. When I pick a file, it's opened in the last split I was focused on.

What actually happens (bold doesn't match my expectations):

1. Ag.vim opens a quickfix window on the bottom of the current window.
1. **Ag.vim opens the first result in the last split I was focused on.**
1. **When I pick a file, it's opened in a seemingly random split that already existed.**

When I run `:LAg <search term>`, I expect the same behavior as `:Ag <search term>`, except in a location list instead of a quickfix window. But here's what happens (bold again for an expectation mismatch):

1. Ag.vim opens a location list on the bottom of the current window.
1. **Ag.vim opens the first result in the last split I was focused on**. 
1. When I pick a file, it's opened in the last split I was focused on.

**Comment by losingkeys** [comment]
Hmm... ag.vim uses [`:grep`](http://vimdoc.sourceforge.net/htmldoc/quickfix.html#:grep) internally. Do you mind doing some tests using `:grep` (you might need to do `:copen` manually after searching) with split windows and see if you get the same results?

We only try to force where things are opened when you're using [the mappings this plugin provides](https://github.com/rking/ag.vim#keyboard-shortcuts) (and, if you look at the issues list, some of those have existing bugs).

**Comment by exchgr** [comment]
When I used `:grep`, I did have to use `:copen` to show the Quickfix window. I had to prefix it with `botright` to get it to open in the same position that `:Ag` opens it. It seems to be displaying the same behavior that I reported.

**Comment by losingkeys** [comment]
That makes sense. You could use :Ag! Too avoid jumping to matches right after searching, then one if the mappings (see :h ag) as a workaround.

Other than that we could try to force it to open splits in a more predictable way, but inertial commands should be predictable. I'll look into the :grep docs to see if I can figure anything out.

**Comment by sinapinto** [comment]
:Ag! jumps to a match right after searching for me

**Comment by cumbreras** [comment]
@exchgr, Did you find a solution for this? I'm experiencing the same issue as you did. Mine opens in a new tab. Unfortunately, I can't find a setting for it on the docs.

**Comment by exchgr** [comment]
@cumbreras Nope. I've mostly stopped using this plugin because of this issue.

**Comment by cumbreras** [comment]
@exchgr What a shame. Thanks for letting me know :+1:

**Comment by losingkeys** [comment]
Hmm, what if you tried adding:
```viml
let g:ag_qhandler="copen"
let g:ag_lhandler="lopen"
```
to your vimrc? I can't recall why these variables were created, but I think they've been there since this plugin was forked from ack.vim. If this works, maybe it's time to change the defaults.

**Comment by typeoneerror** [comment]
Is there any workaround for stopping the selection of the first result in the search and opening that file? The way it's currently configured, you have to open a new tab to not lose your current working file.

**Comment by exchgr** [comment]
I just tried getting this to work again, and I had some success jumping off @losingkeys's suggestion. Here's how I matched my expectations as outlined in my comment on Feb 13:

In vimrc:
```viml
let g:ag_lhandler="botright lopen"
```

Then, running `:LAg! <search-query>` behaves as expected.

**Comment by losingkeys** [comment]
@typeoneerror Mind opening an issue for this option? I thought about it before... but it seemed like it was going against vim's convention, but if people will use it I'm not opposed to adding it.

@exchgr strange. Does 
```viml
let g:ag_lhandler="lopen"
```
Work? The `botright` part is just so new windows open in the bottom/right part of vim. If we're reusing windows it shouldn't be necessary. I'll have to look up why vim's resuing windows here and why ag.vim defaults options to having "botright" in them, it doesn't seem (to me) like a good default.

**Comment by exchgr** [comment]
`let g:ag_lhandler="lopen"` works, but it opens a split under the current window. That's fine, but I was just trying to mimic the default behavior somewhat. Without either of those options, `Ag` opens a quickfix window on the bottom right. That also happens to be my preference.

**Comment by losingkeys** [comment]
@exchgr is this no longer an issue then? Sorry for the delay, I'm just now going through the issues again. A quick scan of this one looks like you have it working how you want, but we/I should make sure this plugin respects options like `splitbelow` and `splitright`and maybe change the `g:ag_*handler` options to default to whatever those are (e.g. remove any `:vertical` or `:botright` prefixes from them).

**Comment by exchgr** [comment]
@losingkeys What are you saying, change the default behavior (or add the ability to do so from a `.vimrc`) if the plugin respects those options?

**Comment by losingkeys** [comment]
I'm saying maybe change this plugin's default behavior if it overrides vim's default, to avoid surprises.
